### PR TITLE
[1LP][RFR] Extract ApplianceConsole and fixtures cli methods into cfme.utils.appliance.console

### PR DESCRIPTION
**Pull request by jaryn**
## Purpose or Intent
Hopefully an __enhancement__ of how the code is layed out in the repo.

{ py.test: -v cfme/tests/cli/test_appliance_console_db_restore.py }

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 97de30e8690d0851f024ab35e3bce30348e81536

The *global* fixture **`ha_appliances_with_providers`** is used in the following files:
- **cfme/tests/cli/test_appliance_update.py**
    - *test_update_ha*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit ec9f30a4876856227a4871f6df194581697eefac

The *global* fixture **`ha_appliances_with_providers`** is used in the following files:
- **cfme/tests/cli/test_appliance_update.py**
    - *test_update_ha*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/utils/appliance/console.py [hunk 1]
```diff
@@ -0,0 +1,428 @@
+import os
+import socket
+import tempfile
+from contextlib import contextmanager
+from re import escape as resc
```
Comment (mshriver) [inline cfme/utils/appliance/console.py]: Personally I don't really like renaming this import, thoughts @ManageIQ/qe-committers?

Comment (jaryn) [reply cfme/utils/appliance/console.py]: I don't like that it seems like some "rscue" would be done. I was not able to find out better name that would still be short for often use. Therefore I suggest using just `esc()` or just `r()`?

Comment (jaryn) [reply cfme/utils/appliance/console.py]: `ree()` ?

**Comment by dajoRH** [comment]
I detected some fixture changes in commit bd39696e508f4bc14a53a5dbe4d4b6682fdfc718

The *global* fixture **`ha_appliances_with_providers`** is used in the following files:
- **cfme/tests/cli/test_appliance_update.py**
    - *test_update_ha*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 4eee9a114cc8af22a953ed70f34004a209fede80

The *global* fixture **`ha_appliances_with_providers`** is used in the following files:
- **cfme/tests/cli/test_appliance_update.py**
    - *test_update_ha*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/utils/appliance/console.py (hunk 1 above)
Comment (mshriver) [reply cfme/utils/appliance/console.py]: I would just `import re`, and call `re.escape`

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 9f8c3c822eace201769f208ce0e28680f1556ebd

The *global* fixture **`ha_appliances_with_providers`** is used in the following files:
- **cfme/tests/cli/test_appliance_update.py**
    - *test_update_ha*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 03739eb21c4c68e6e592373d5008968dfce08a9f

The *global* fixture **`ha_appliances_with_providers`** is used in the following files:
- **cfme/tests/cli/test_appliance_update.py**
    - *test_update_ha*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 3adba49c5d56ff20d817a2e3392ccd673ed4592e

The *global* fixture **`ha_appliances_with_providers`** is used in the following files:
- **cfme/tests/cli/test_appliance_update.py**
    - *test_update_ha*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/utils/appliance/console.py [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+            interaction.answer(resc('Apply this Replication Server Configuration? (Y/N): '), 'y')
+            interaction.answer('Press any key to continue.', '')
+
+    def configure_standby_replication_node(self, pwd, primary_ip):
+        # Configure secondary (standby) replication node
+        with SSHExpect(self.appliance) as interaction:
+            interaction.send('ap')
+            interaction.answer('Press any key to continue.', '', timeout=20)
+            interaction.answer('Choose the advanced setting: ',
+                               '6' if self.appliance.version < '5.10' else '8')
```
Comment (jawatts) [inline cfme/utils/appliance/console.py]: **Optional** since it doesn't change functionality, but we can remove this check.

**Comment by jawatts** [comment]
Nice work Jaro!!

File: cfme/utils/appliance/console.py (hunk 2 above)
Comment (jaryn) [reply cfme/utils/appliance/console.py]: Umn. Quite a good point, but I am also doing upgrade testing and there are some not-automated test doing upgrades on ha setup. There is `test_upgrade_multi_ha_inplace` which needs to be automated and once I do that, it certainly will use this method and the condition will come handy.

**Comment by jaryn** [comment]
> Nice work Jaro!!

Yeah. Especially when the result is 12 lines shorter than the original.

**Pull request merged**